packages/maskbook/src/plugins/RedPacket/database.ts

```diff
@@ -10,6 +10,14 @@ export function getRedPackets() {
     return asyncIteratorToArray(RedPacketDatabase.iterate('red-packet'))
 }
 
+export async function getRedPacketsHistory(rpids: string[]) {
+    const records: RedPacketRecord[] = []
+    for await (const record of RedPacketDatabase.iterate('red-packet')) {
+        if (rpids.includes(record.payload.rpid)) records.push(RedPacketRecordOutDB(record))
```

Comment:
is it possible to have multiple records with the same rpid?

Reply:
No, each record has its only rpid.

Reply:
so why don't you get record by `RedPacketDatabase.get`?

Reply:
the last question @zhouhanseng

Reply:
if you can query via `.get` plz use it instead of iterate over the database. it's O1 (hash table)

Reply:
cc @zhouhanseng